### ISSUE TITLE
[BUG] Cannot acquire LWA OAuth Code, documentation does not help

### ISSUE THREAD

**Issue by michu94** [amzn/selling-partner-api-docs#2150]
Hello,
I’m trying to migrate my Developer application to new SP Api. Currently I’m targeting all of my requests against old MWS Api. Since Amazon is shutting down some of the endpoints in foreseeable future I’m forced to comply with new SP Api standard.  

Even though I tried to follow all the instructions given in SP-API docs (https://developer-docs.amazon.com/sp-api/) I still cannot successfully migrate my requests. I cannot get LWA OAuth code which I need to acquire refresh token for my client’s requests.

These are steps which I’ve taken so far:
1)	Configured my application on Amazon Seller Central to be a hybrid application (so I can ingest both SP & MWS Api). 
2)	I confirmed that my application is fully published on Amazon.
3)	I’ve created a AWS role and a user on behalf of which I will be making my requests.
4)	I called this endpoint to assume IAM role:
 
![image](https://user-images.githubusercontent.com/18336890/159696149-900c4264-5b95-4640-ae4f-85d6f978b191.png)

5)	I called this endpoint to acquire LWA Access token:
 
![image](https://user-images.githubusercontent.com/18336890/159696165-21390c22-2e74-4b32-ad3f-f38ee8641f17.png)

6)	I tried to call this endpoint to acquire LWA OAuth Code:
 
![image](https://user-images.githubusercontent.com/18336890/159696180-70b027d3-ee11-4553-b8ab-fe1d97254acb.png)

And received such a response:

 
![image](https://user-images.githubusercontent.com/18336890/159696191-9b49bd29-bfd6-4b52-b4cc-beebd6e2e276.png)

All of my credentials seems okey and this request should go through. BUT IT DOES NOT FOR SOME REASON… Error message does not say anything and response code is 500 – which indicates that there is INTERNAL SERVER ERROR on Amazon side.

In response headers I have followings values:
x-amzn-RequestId : 004d4be6-3265-4d86-a830-94769b4d0145
x-amz-apigw-id : Pb7A3FisDoEFXRw=
X-Amzn-Trace-Id : Root=1-623b0b9e-30a3065b510a23a61fa17ad3

I hope those will help you to find me an answer why I cannot obtain LWA OAuth Code.

Can you help me figure out what is wrong with my approach? From my perspective everything should be working but as you can see, it does not ☹

I would appreciate all the help I can get.

Enjoy you day.
Best regards.

**Comment by lobotomisador** [comment]
@michu94 are you signing the request to the authorizationCode endpoint? It requires a specific set of headers, see :https://docs.aws.amazon.com/general/latest/gr/signing_aws_api_requests.html. This is very cumbersome to do manually, I would suggest using an official aws SDK for the language you are using. When you do this, log the response `requestId` header so you can conctact aws customer support and they can pinpoint your issue.

**Comment by KyleJune** [comment]
@michu94 We were having the same issue up until today. If your application was published before you changed it to being a hybrid app, it still shows as published after you make that change. However that change doesn't actually go into affect unless you re-publish the application. For us, it took 8 days from the time we submitted for republishing before it was published. We continued getting the error up until the moment our application started showing as published again.

It would be helpful if they gave an error with a more useful message like "Application not published as a hybrid application. It may need republishing if it was published before changing it to be a hybrid application".
